### DISCUSSION TITLE
"Failed to look up rubygems package sidekiq-pro" is back

### DISCUSSION THREAD

**Discussion by mscrivo** [renovatebot/renovate#22896] [Request Help]
### How are you running Renovate?

Mend Renovate hosted app on github.com

### If you're self-hosting Renovate, select which platform you are using.

None

### Was this something which used to work for you, and then stopped?

It used to work, and then stopped

### Describe the problem

It seems that this recently fixed [issue](https://github.com/renovatebot/renovate/issues/16126) is back as of a few days ago. We are seeing this in our dependency dashboard again:

<img width="845" alt="image" src="https://github.com/renovatebot/renovate/assets/275524/64134955-d993-46bc-8372-c6932c68fa9c">

Perhaps it's due to the changes in [this PR](https://github.com/renovatebot/renovate/pull/22860) (which we were very happy to see finally fixed btw!)?

Here's what we have in our renovate.json (which has not changed in a long time):
```
  "hostRules": [
    {
      "matchHost": "https://gems.contribsys.com/",
      "hostType": "rubygems",
      "username": "redacted",
      "password": "redacted"
    }
  ],
```

Providing a minimal repro requires having a sidekiq-pro license, so I have not included one, but it's pretty much just adding the above to your renovate.json.

### Relevant debug logs

<details><summary>Logs</summary>

```
DEBUG: GET https://gems.contribsys.com/api/v1/versions/sidekiq-pro.json = (code=ERR_NON_2XX_3XX_RESPONSE, statusCode=404 retryCount=0, duration=1031)
DEBUG: Datasource 404
{
  "datasource": "rubygems"
  "packageName": "sidekiq-pro"
  "url": "https://gems.contribsys.com/api/v1/versions/sidekiq-pro.json"
}

DEBUG: Failed to look up rubygems package sidekiq-pro
{
  "dependency": "sidekiq-pro"
  "packageFile": "Gemfile"
}
```

</details>

### Have you created a minimal reproduction repository?

I have explained in the description why a minimal reproduction is impossible

**Comment by rarkins** [comment]
Is the URL correct/same as before?

**Reply by mscrivo** [reply]
I don't know how to check if it was the same as before, but https://gems.contribsys.com/info/sidekiq-pro does work, whereas https://gems.contribsys.com/api/v1/versions/sidekiq-pro.json 404's and so it seems the fallback to the old style "versions" endpoint is no longer working?

**Reply by rarkins** [reply]
Can you set up a public reproduction without credentials?

**Reply by mscrivo** [reply]
done https://github.com/mscrivo/renovate-sidekiq-pro

**Reply by rarkins** [reply]
URLs today:

```
DEBUG: http statistics (repository=renovate-reproductions/22896)
       "urls": {
         "https://api.github.com/graphql (POST,200)": 1,
         "https://api.github.com/repos/renovate-reproductions/22896/pulls (GET,200)": 1,
         "https://gems.contribsys.com/api/v1/versions/sidekiq-pro.json (GET,404)": 1,
         "https://gems.contribsys.com/versions (GET,200)": 1,
         "https://www.ruby-lang.org/en/downloads/releases/ (GET,200)": 1
       },
```

URLs from older release:

```
DEBUG: http statistics (repository=renovate-reproductions/22896)
       "urls": {
         "https://api.github.com/graphql (POST,200)": 1,
         "https://api.github.com/repos/renovate-reproductions/22896/pulls (GET,200)": 1,
         "https://gems.contribsys.com/versions (GET,206)": 1,
         "https://www.ruby-lang.org/en/downloads/releases/ (GET,200)": 1
       },
```

The change is `"https://gems.contribsys.com/api/v1/versions/sidekiq-pro.json (GET,404)": 1,`. @zharinov should it be an optional URL?
